Gems/AtomLyIntegration/AtomViewportDisplayInfo/Code/Source/AtomViewportDisplayInfoSystemComponent.cpp

```diff
@@ -256,6 +258,64 @@ namespace AZ::Render
         m_fpsHistory.push_back(currentTime);
     }
 
+    void AtomViewportDisplayInfoSystemComponent::DrawMemoryInfo()
+    {
+        RHI::RHISystemInterface* rhi = RHI::RHISystemInterface::Get();
+        if (!rhi)
+        {
+            return;
+        }
+
+        const RHI::MemoryStatistics* stats = rhi->GetMemoryStatistics();
+        if (!stats)
+        {
+            return;
+        }
+
+        // Accumulate total device memory pressure (reserved, resident)
+        size_t deviceResident = 0;
+        size_t deviceReserved = 0;
+
+        for (const auto& pool : stats->m_pools)
+        {
+            deviceReserved += pool.m_memoryUsage.GetHeapMemoryUsage(RHI::HeapMemoryLevel::Device).m_reservedInBytes;
+            deviceResident += pool.m_memoryUsage.GetHeapMemoryUsage(RHI::HeapMemoryLevel::Device).m_residentInBytes;
+        }
+
+        // Query for available device memory
+        float availableDeviceMemoryMB = 0.f;
+
+        static constexpr size_t MB = 1u << 20;
+
+        if (RHI::Device* device = rhi->GetDevice(); device)
```

Comment:
`if (RHI::Device* device = rhi->GetDevice())` is enough here to get device checked

Reply:
Agreed and it's probably what I would write in my own code, but the scoped initializer style is a bit more common in this codebase, so I'm content to adopt the convention-apparent as it were.